typescript errors as of 0.2.75

**Describe the bug**
The jsdoc types added to the `is` functions in 0.2.75 export typescript definitions which do not typecheck

**To Reproduce**
`npm run clean && npm run types && npm run lint` yields the following errors
```
function.d.ts:12:62 - error TS2344: Type 'TYPE' does not satisfy the constraint 'abstract new (...args: any) => any'.

12 export function is<TYPE>(n: any, T: TYPE): n is InstanceType<TYPE>;
                                                                ~~~~

  function.d.ts:12:20
    12 export function is<TYPE>(n: any, T: TYPE): n is InstanceType<TYPE>;
                          ~~~~
    This type parameter might need an `extends abstract new (...args: any) => any` constraint.

function.d.ts:13:74 - error TS2344: Type 'TYPE' does not satisfy the constraint 'abstract new (...args: any) => any'.

13 export function isTemplate<TYPE>(T: TYPE): (n: any) => n is InstanceType<TYPE>;
                                                                            ~~~~

  function.d.ts:13:28
    13 export function isTemplate<TYPE>(T: TYPE): (n: any) => n is InstanceType<TYPE>;
                                  ~~~~
    This type parameter might need an `extends abstract new (...args: any) => any` constraint.

isomorphic.d.ts:1:10 - error TS2303: Circular definition of import alias 'performance'.

1 export { performance, cryptoRandomBuffer } from "isomorphic.js";
           ~~~~~~~~~~~

isomorphic.d.ts:1:23 - error TS2303: Circular definition of import alias 'cryptoRandomBuffer'.

1 export { performance, cryptoRandomBuffer } from "isomorphic.js";
                        ~~~~~~~~~~~~~~~~~~
```

**Expected behavior**
Lints pass

**Environment Information**
 - node v18.16.0
 
 
 **Additional Information**
 you may want to add the linting as a prepublish script

--- a/function.js
+++ b/function.js
@@ -167,7 +167,7 @@ export const isString = (s) => s && s.constructor === String
 export const isNumber = n => n != null && n.constructor === Number
 
 /**
- * @template TYPE
+ * @template {abstract new (...args: any) => any} TYPE
  * @param {any} n
  * @param {TYPE} T
  * @return {n is InstanceType<TYPE>}
@@ -175,7 +175,7 @@ export const isNumber = n => n != null && n.constructor === Number
 export const is = (n, T) => n && n.constructor === T
 
 /**
- * @template TYPE
+ * @template {abstract new (...args: any) => any} TYPE
  * @param {TYPE} T
  */
 export const isTemplate = (T) =>

